pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -985,6 +998,17 @@ func (r *rangeKeyBatcher) flush(ctx context.Context, toFlush mvccRangeKeyValues)
 func splitRangeKeySSTAtKey(
 	ctx context.Context, st *cluster.Settings, start, end, splitKey roachpb.Key, data []byte,
 ) (*rangeKeySST, *rangeKeySST, error) {
+	// Special case: The split key less than the start key.
+	if splitKey.Compare(start) < 0 {
+		return nil, &rangeKeySST{start: start, end: end, data: data}, nil
+	}
+
+	// Special case: The split key is greater or equal to the
+	// exclusive end key.
+	if end.Compare(splitKey) <= 0 {
+		return &rangeKeySST{start: start, end: end, data: data}, nil, nil
+	}
```

Comment:
I could be convinced that these should just also return errors since they really should not happen.

Reply:
I think I was just not thinking this through when I wrote this.  I think what I had in my mind is that if the whole SST was in a different range, then why did we ever send it to the wrong range.  But, perhaps our entire SST is on one side of a split we don't know about yet.  So, I'll leave these cases and remove the errant comment.

Thanks!